alibi_detect/saving/schemas.py

```diff
@@ -79,6 +79,33 @@ def validate_optimizer(cls, optimizer: Any, values: dict) -> Any:
 #  of preprocess_drift.
 
 
+class SupportedDevice:
+    """
+    Pydantic custom type to check the device is correct for the choice of backend (conditional on what optional deps
+    are installed).
+    """
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate_device
+
+    @classmethod
+    def validate_device(cls, device: Any, values: dict) -> Any:
+        backend = values['backend']
+        if backend == Framework.TENSORFLOW or backend == Framework.SKLEARN:
+            if device is not None:
+                raise TypeError('`device` should not be specified for TensorFlow or Sklearn models. Leave as `None`.')
```

Comment:
Nitckpick: `TensorFlow or SKlearn models` - "models" is maybe a little inaccurate, since we are talking about detector backends here?